examples/chart/teleport-cluster/templates/deployment.yaml

```diff
@@ -251,10 +251,10 @@ spec:
         secret:
           secretName: "license"
 {{- end }}
-{{- if (eq .Values.chartMode "gcp") }}
+{{- if and (.Values.gcp.credentialSecretName) (eq .Values.chartMode "gcp")  }}
```

Comment:
`credentialSecretName` is still listed as required in the schema.

Additionally, it would be good to have a test for this.

Reply:
Thanks @zmb3 - I've updated the schema to remove the required value.

With the tests, this is a fix for a breaking change that was introduced, rather than new functionality, so I'm not too sure what can be tested here? More than happy to add one if you're able to advise?

Reply:
In this case, you should be able to write a test that does _not_ set `gcp.credentialSecretName` and verifies that the `gcp-credentials` secret is not rendered.

Looks like `gcp-ha.yaml` is not setting that value, so you should be able to use that one. Take a look at `examples/chart/teleport-cluster/tests/deployment-test.yaml`.

Something along these lines should do, though it's been a while since I've looked at the various assertions:

```
  - it: does not render the GCP credentials secret when not provided
    template: deplyoment.yaml
    values:
    - ../.lint/gcp-ha.yaml
    asserts:
    - ...
```

Reply:
That makes sense, I've added a test for this usecase. Needed to add an additional lint file for this specific test since `gcp.credentialSecretName` requires a blank value when not required (as per the [value docs](https://github.com/gravitational/teleport/blob/master/examples/chart/teleport-cluster/values.yaml#L239)).

Tests all pass with this addition;

```
### Chart [ teleport-cluster ] .

 PASS  Certificate	tests/certificate_test.yaml
 PASS  ClusterRole	tests/clusterrole_test.yaml
 PASS  ClusterRoleBinding	tests/clusterrolebinding_test.yaml
 PASS  ConfigMap	tests/config_test.yaml
 PASS  Deployment	tests/deployment_test.yaml
 PASS  PodDisruptionBudget	tests/pdb_test.yaml
 PASS  PodSecurityPolicy	tests/psp_test.yaml
 PASS  PersistentVolumeClaim	tests/pvc_test.yaml
 PASS  Service	tests/service_test.yaml
 PASS  ServiceAccount	tests/serviceaccount_test.yaml

Charts:      1 passed, 1 total
Test Suites: 10 passed, 10 total
Tests:       127 passed, 127 total
Snapshot:    120 passed, 120 total
Time:        724.917292ms
```

Reply:
Nice, thanks @adamstrawson! I also verified that the test fails without your change. Appreciate you taking the time to add this.